File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/dto/AssetHistoryReq.java
package com.NBE3_4_2_Team4.domain.asset.main.dto;

import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetCategory;
import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetType;
import com.NBE3_4_2_Team4.domain.asset.main.validation.annotation.ValidAssetHistoryReq;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@ValidAssetHistoryReq
@Getter
@Setter
public class AssetHistoryReq {
    @NotNull
    @Min(1)
    private Integer page=1;

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate startDate;

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate endDate;

    private AssetCategory assetCategory;

    private AssetType assetType;

    public LocalDateTime getStartDateTime() {
        return startDate != null ? startDate.atStartOfDay() : null;
    }

    public LocalDateTime getEndDateTime() {
        return endDate != null ? endDate.atTime(LocalTime.MAX) : null;
    }

}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/dto/AssetHistoryReq.kt
package com.NBE3_4_2_Team4.domain.asset.main.dto

import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetCategory
import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetType
import com.NBE3_4_2_Team4.domain.asset.main.validation.annotation.ValidAssetHistoryReq
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

@ValidAssetHistoryReq
data class AssetHistoryReq(
    @field:NotNull
    @field:Min(1)
    val page: Int = 1,

    val assetType: AssetType? = null,

    val assetCategory: AssetCategory? = null,

    @field:DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    val startDate: LocalDate? = null,

    @field:DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    val endDate: LocalDate? = null
) {
    @get:JsonIgnore
    val startDateTime: LocalDateTime?
        get() = startDate?.atStartOfDay()

    @get:JsonIgnore
    val endDateTime: LocalDateTime?
        get() = endDate?.atTime(LocalTime.MAX)
}
